pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -415,6 +454,60 @@ def queue_view(self):
         """
         return self.view_queues()
 
+    @property
+    def executor(self) -> Union[Executor, None]:
+        """
+        Executor to execute the job object this server object is attached to in the background.
+
+        Returns:
+            concurrent.futures.Executor:
+        """
+        if not self.run_mode.executor and self._executor is not None:
+            self._executor = None
+        return self._executor
+
+    @executor.setter
+    def executor(self, exe: Union[Executor, None]):
+        """
+        Executor to execute the job object this server object is attached to in the background.
+
+        Args:
+            exe (concurrent.futures.Executor):
+        """
+        if isinstance(exe, Executor):
+            self.run_mode.executor = True
+        elif exe is None:
+            self.run_mode.modal = True
```

Comment:
I added an alternative suggestion above - only switch the `run_mode` to `modal` when the `run_mode` is still set to `executor` otherwise the `run_mode` is not modified.

Reply:
> I added an alternative suggestion above 

I don't see this, but that sounds fine too. It _should_ be formally equivalent, as any time `server.executor is not None` I expect `server.run_mode.executor`. Anything that stops a None-executor being set to more None from updating the run mode will satisfy me.

Reply:
Ah, I see it. It was just "below" for me and not "above". Hard to read with all the layered "suggestion" bits, but what you describe here sounds fine and I trust that's what the suggestions result in.